be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -872,13 +867,13 @@ Status SegmentIterator::next_batch(vectorized::Block* block) {
         RETURN_IF_ERROR(_init(true));
         _inited = true;
         if (!_vec_pred_column_ids.empty() || !_short_cir_pred_column_ids.empty()) {
-            _block_rowids.reserve(_opts.block_row_max);
+            _block_rowids.resize(_opts.block_row_max);
```

Comment:
When first to init SegmentIterator, _block_rowids needs to reserve;
Because _block_rowids is accessed by operator[] directly

Reply:
reverve just alloc memory for optimization reason.

there is one basic rule: any reserve calls can be removed, this removing operation will not influence the correctness of result.

according this rule, if we remove this _block_rowids.reserve(), the will crash, so we should use resize instead.